purchase_stock_picking_invoice_link/models/account_invoice.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2019 Vicent Cubells <vicent.cubells@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    def _update_picking_invoices(self):
+        self.ensure_one()
+        purchase = self.invoice_line_ids.mapped('purchase_line_id.order_id')
+        if not purchase:
+            return
+        self.mapped('invoice_line_ids.move_line_ids').filtered(
+            lambda x: x.state == 'done'
+            and not x.location_dest_id.scrap_location
+            and x.location_id.usage == 'supplier').mapped(
+                'picking_id').write({'invoice_ids': [(4, self.id)]})
```

Comment:
@cubells Are you sure of that ??? It could be bulky in terms of performance and I think this is not a good idea to call a write on onchange.